test/api/AssetTest.ts

```diff
@@ -320,6 +363,82 @@ describe('Asset', function() {
         );
       });
     });
+
+    describe('IoThink Asset Connector tests', function() {
+
+      before(async () => {
+        // Check if ioThink asset connector parameters are set
+        if (!config.get('assetConnectors.ioThink.user') ||
+          !config.get('assetConnectors.ioThink.password') ||
+          !config.get('assetConnectors.ioThink.meterID')) {
+          this.pending = true;
+        }
+        testData.userContext = testData.tenantContext.getUserContext(ContextDefinition.USER_CONTEXTS.DEFAULT_ADMIN);
+        if (testData.userContext === testData.centralUserContext) {
+          // Reuse the central user service (to avoid double login)
+          testData.userService = testData.centralUserService;
+        } else {
+          testData.userService = new CentralServerService(
+            testData.tenantContext.getTenant().subdomain,
+            testData.userContext
+          );
+        }
+        // Create a new Asset
+        testData.newAsset = await testData.userService.createEntity(
+          testData.userService.assetApi,
+          Factory.asset.build({
+            siteAreaID: testData.newSiteArea.id,
+            assetType: 'PR',
+          })
+        );
+        testData.createdAssets.push(testData.newAsset);
+        expect(testData.newAsset).to.not.be.null;
+      });
+
+      it('Should be able to create an ioThink asset connector', async function() {
```

Comment:
This test is using different functionality, if it is running the first time after context creation.
The creation of the asset connector is only tested in the first run. If your run the tests more than once updating the connector is checked.

Reply:
The key used to access the token cache in IoThinkAssetIntegration as well as WitAssetIntegration had been changed to include user and encrypted password along with original connection id. This way, if the details such as username or password is changed, a new token is fetched again instead of reusing existing tokens.

Reply:
Correct, but the test case says 'Should be able to create an ioThink asset connector'. 
But it is only created in the first test run, in the second it is updated. 
This has nothing to do with the token.

Reply:
Added the changes to delete the settings once the asset connector tests are executed.